src/Init/Data/Array/Basic.lean

```diff
@@ -50,6 +50,13 @@ instance : Inhabited (Array α) where
 def singleton (v : α) : Array α :=
   mkArray 1 v
 
+/-- Low-level version of `size` that directly queries the C array object cached size.
+    While this is not provable, `usize` always returns the exact size of the array since
+    the implementation only supports arrays of size less than `USize.size`.
+-/
+@[extern "lean_array_size", simp]
+def usize (a : @& Array α) : USize := a.size.toUSize
```

Comment:
It seems that other functions from C world that are used via `extern` have a
`LEAN_EXPORT` declaration in the corresponding header file. Did you intentionally not add one of these for the new functions?

Reply:
I think `LEAN_EXPORT` only matters for symbols exported to a shared library. That doesn't seem necessary in this case. For example, `lean_array_get_size` (which implements `Array.size`) does not have a `LEAN_EXPORT`.

Reply:
Great, thanks for checking.